src/figuresClasses.ts

```diff
@@ -1,11 +1,77 @@
-export interface Figure {}
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+export class Triangle implements Figure {
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+    public shape = 'triangle',
```

Comment:
[CODE STYLE] - Enum constant names usually start with a capital letter. While TypeScript enums are not used here, the principle remains the same for default values of the 'shape' property. Consider using 'Triangle' instead of 'triangle'.